src/main/java/io/curity/identityserver/plugin/nonce/anonymous/JwtValidator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright 2025 Curity AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.curity.identityserver.plugin.nonce.anonymous;
+
+import io.curity.identityserver.plugin.nonce.config.NonceAuthenticatorPluginConfig;
+import org.jose4j.http.SimpleGet;
+import org.jose4j.http.SimpleResponse;
+import org.jose4j.jwk.HttpsJwks;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwt.consumer.JwtContext;
+import org.jose4j.keys.resolvers.HttpsJwksVerificationKeyResolver;
+import se.curity.identityserver.sdk.http.HttpResponse;
+import se.curity.identityserver.sdk.plugin.ManagedObject;
+import se.curity.identityserver.sdk.service.HttpClient;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * JwtValidator is a ManagedObject to be able to be the validation keys stored, avoiding downloading them for every
+ * request. Keys will be updated if a key is not found
+ */
+public final class JwtValidator extends ManagedObject<NonceAuthenticatorPluginConfig>
```

Comment:
We should really create that library to avoid copying this around

Reply:
noted.